src/main/kotlin/io/github/inductiveautomation/kindling/gatewaynetwork/GatewayNetworkViewer.kt

```diff
@@ -0,0 +1,161 @@
+                    val theText = textArea.text
+
+                    // Verify that the provided text is valid JSON, as the user is free to copy and paste in text.
+                    try {
+                        JSONObject(theText)
+                    } catch (e: JSONException) {
+                        JOptionPane.showMessageDialog(
+                            null,
+                            "Error: " + e.message,
+                            "JSON parsing error",
+                            JOptionPane.ERROR_MESSAGE,
+                        )
+                        return
+                    }
+                    JSONObject(theText)
+
+                    val tmpDir: String = writeStaticFiles(theText)
+
+                    val tempAddress = "file://$tmpDir/index.html"
+                    val desktop = if (Desktop.isDesktopSupported()) Desktop.getDesktop() else null
+                    if (desktop != null && desktop.isSupported(Desktop.Action.BROWSE)) {
+                        try {
+                            desktop.browse(URI(tempAddress))
+                        } catch (error: Exception) {
+                            JOptionPane.showMessageDialog(
+                                null,
+                                "Error: " + error.message,
+                                "Error opening browser",
+                                JOptionPane.ERROR_MESSAGE,
+                            )
+                        }
+                    }
+                }
+            },
+        )
+
+        // Temp file cleanup when Kindling exits.
+        val shutdownThread = Thread {
+            val tmpDir: File = Path.of(System.getProperty("java.io.tmpdir"), "gateway-network-diagram").toFile()
+            tmpDir.deleteRecursively()
+        }
+
+        Runtime.getRuntime().addShutdownHook(shutdownThread)
+    }
+
+    companion object {
+        private val FAVICON = "favicon.ico"
+        private val INDEX_HTML = "index.html"
+        private val MAIN_JS = "main.js"
+        private val STYLE_CSS = "style.css"
+        private val EXTERNAL_DIAGRAM_JS = "external-diagram.js"
+        private val PREAMBLE = "window.externalDiagram = "
+    }
+
+    private fun parseLines(lines: Sequence<String>, textArea: JTextArea) {
+        val builder = StringBuilder()
+        for (line in lines) {
+            if (line.isBlank()) {
+                continue
+            } else {
+                builder.append(line)
+                builder.append("\n")
+            }
+        }
+        textArea.setText(builder.toString())
+    }
+
+    private fun writeStaticFiles(jsonText: String): String {
+        // Make a temp dir for the static html/js files
+        val tmpDir: Path = Path.of(
+            System.getProperty("java.io.tmpdir"),
+            "gateway-network-diagram",
+            UUID.randomUUID().toString(),
+        )
+        Files.deleteIfExists(tmpDir)
+        Files.createDirectories(tmpDir)
+
```

Comment:
There's a `createTempDirectory` builtin, e.g.
```suggestion
createTempDirectory("kindling-gateway-network-diagram")
```